framework/src/play-server/src/main/scala/play/core/server/Server.scala

```diff
@@ -89,30 +89,33 @@ object Server {
    */
   private[server] def getHandlerFor(request: RequestHeader, tryApp: Try[Application]): (RequestHeader, Handler) = {
 
-    @inline def handleErrors(errorHandler: HttpErrorHandler): PartialFunction[Throwable, (RequestHeader, Handler)] = {
+    @inline def handleErrors(errorHandler: HttpErrorHandler, req: RequestHeader): PartialFunction[Throwable, (RequestHeader, Handler)] = {
       case e: ThreadDeath => throw e
       case e: VirtualMachineError => throw e
       case e: Throwable =>
-        val errorResult = errorHandler.onServerError(request, e)
+        val errorResult = errorHandler.onServerError(req, e)
         val errorAction = actionForResult(errorResult)
         (request, errorAction)
```

Comment:
Should probably use `req` too:
```suggestion
         (req, errorAction)
```

Reply:
Hmm, maybe not?  If not, maybe it should be called out in the code (i.e add a comment.)

Reply:
Marcos gave me some explanations about this fix yesterday, I think that the data that is missing and is causing the bug is needed when producing the `errorResult`, that's all. 
That's why we use it to produce the error, but we use the original request to return. 

For the sake of simplicity, I would prefer to return `req` (the enriched request) everywhere. Just to avoid extra brain cycles in understanding the code. By the way, we should rename it to `enrichedRequest` because that's what it is.

Reply:
`req` is only the enriched request on the first invocation:

https://github.com/marcospereira/playframework/blob/fix-8947-webcommands-fail-to-copy-request/framework/src/play-server/src/main/scala/play/core/server/Server.scala#L115

The second time it's the original request and we don't have access to the enriched request:

https://github.com/marcospereira/playframework/blob/fix-8947-webcommands-fail-to-copy-request/framework/src/play-server/src/main/scala/play/core/server/Server.scala#L118

Reply:
The second time is when the application fails to initialize so that we won't have an enriched request. It is also okay to use the original one since the application error handler won't kick in.

Reply:
But, yeah, we should use `req` here consistently.